integration-tests/mybatis/src/test/java/org/apache/camel/quarkus/component/mybatis/it/MyBatisTest.java

```diff
@@ -70,17 +98,126 @@ public void testInsert() {
                 .body(equalTo("3"));
     }
 
+    public void testInsertRollback() {
+        Account account = new Account();
+        account.setId(999);
+        account.setFirstName("Rollback");
+        account.setLastName("Rollback");
+        account.setEmailAddress("Rollback@gmail.com");
+
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body(account)
+                .post("/mybatis/insertOne")
+                .then()
+                .statusCode(500);
+    }
+
+    public void testInsertList() {
+        Account account1 = new Account();
+        account1.setId(555);
+        account1.setFirstName("Aaron");
+        account1.setLastName("Daubman");
+        account1.setEmailAddress("ReadTheDevList@gmail.com");
+
+        Account account2 = new Account();
+        account2.setId(666);
+        account2.setFirstName("Amos");
+        account2.setLastName("Feng");
+        account2.setEmailAddress("ZHENG@gmail.com");
+
+        List<Account> accountList = new ArrayList<>(2);
+
+        accountList.add(account1);
+        accountList.add(account2);
+
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body(accountList)
+                .post("/mybatis/insertList")
+                .then()
+                .statusCode(200)
+                .body(equalTo("5"));
```

Comment:
Why it does return 5 here ? We batch insert only 2 items ? Maybe there 3 before that ?

Reply:
Ok, it's clearer now. It's the kind of integration test scenario where the result of an assert can depend on previous operation. Many thanks of clarification :+1: